src/packages/src/xroad/common/proxy/usr/share/xroad/scripts/restore_db.sh

```diff
@@ -41,13 +41,18 @@ psql_dbuser() {
   PGOPTIONS="$pg_options" PGDATABASE="$db_database" PGUSER="$db_user" PGPASSWORD="$db_password" remote_psql "$@"
 }
 
+pgrestore() {
+PGHOST="$db_addr" PGPORT="$db_port" PGUSER="$db_admin_user" PGPASSWORD="$db_admin_password" \
+  pg_restore --single-transaction -d "$db_database" --schema=$db_schema $dump_file
+}
+
 { cat <<EOF
-BEGIN;
 DROP SCHEMA IF EXISTS "$db_schema" CASCADE;
+CREATE SCHEMA "$db_schema";
 EOF
-  cat "$dump_file"
-  echo "COMMIT;"
-} | psql_adminuser || abort "Restoring database failed."
+} | psql_adminuser || abort "Dropping schema failed."
+
+pgrestore || abort "Restoring database failed."
```

Comment:
The previous implementation did not change the database if restore failed, but now the database can be left in a broken state. One option would be to just create the schema if it does not exist (because pg_restore will not recreate it), and use the --clean option in restore (which runs in a transaction).

Reply:
--clean does not drop objects  that are not in the backup... but then again this is only an issue when backup is made from a different versions - a situation that must be handled separately. So yes, lets go wit IF NOT EXISTS and --clean